playground/terraform/README.md

```diff
@@ -16,74 +16,118 @@
 
-```bash
-$ gsutil mb -p ${PROJECT_ID} gs://state-bucket-name
-$ gsutil versioning set on gs://state-bucket-name
-```
+4. DNS name for your Playground deployment instance
 
-## 1. Create new environment
+***Development environment preparation steps:***
 
-To provide information about the terraform backend, run the following commands
+During the Operation system preparation steps, you will need to install Java, Kubernetes, HELM, Docker, Terraform, gcloud CLI
 
-* New environment folder
+* [Java](https://adoptopenjdk.net/)
+* [Kubernetes Command Line Interface](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/)
+* [HELM](https://helm.sh/docs/intro/install/)
+* [Docker](https://docs.docker.com/engine/install/)
+* [Terraform](https://www.terraform.io/downloads)
+* [gcloud CLI](https://cloud.google.com/sdk/docs/install-sdk)
 
-```bash
-mkdir /path/to/beam/playground/terraform/environment/{env-name}
-```
+After Development environment preparation steps completed, navigate to the "beam" folder in your development environment.
 
-* Backend config
+# Deployment environment:
+Playground uses `terraform.tfvars` located in `playground/terraform/environment/environment_name` to define all infrastructure environment (like staging, production, etc.). Create a folder `environment_name` and place all configuration files inside to add a new environment.
 
-```bash
-echo 'bucket = "put your state bucket name here"' > /path/to/beam/playground/terraform/environment/{env-name}/state.tfbackend
+* `terraform.tfvars` environment variables:
 ```
-
-* Terraform variables config and provide necessary variables
-
-```bash
-touch /path/to/beam/playground/terraform/environment/{env-name}/terraform.tfvars
+project_id           = "project_id"          #GCP Project ID
+network_name         = "network_name"        #GCP VPC Network Name for Playground deployment
+gke_name             = "playground-backend"  #Playground GKE Cluster name
+region               = "us-east1"            #Set the deployment region
+location             = "us-east1-b"          #Select the deployment location from available in the specified region
+state_bucket         = "bucket_name"         #GCS bucket name for Beam Playground temp files, [terraform state file] (https://cloud.google.com/docs/terraform/resource-management/store-state)
+bucket_examples_name = "bucket_name-example" #GCS bucket name for Playground examples storage
 ```
-
-Then provide necessary variables.
-
-## 2. Provision infrastructure
-
-To deploy Playground infrastructure run gradle task:
-
-```bash
-./gradlew playground:terraform:InitInfrastructure -Pproject_environment="env-name"
+* `state.tfbackend` environment variables:
 ```
-
-## 3. Deploy application
-
-To deploy application run following steps:
-
-* Authinticate in Artifact registry
-
-```bash
-gcloud auth configure-docker us-central1-docker.pkg.dev
+bucket               = "bucket_name"         #input bucket name - will be used for terraform tfstate file
 ```
```

Comment:
It's confusing to have variables with different names for the same bucket. If you don't want to change the name of the variable, state explicitly:
Variable 'bucket' in state.tfbackend file should have the same value as 'state_bucket' variable in terraform.tfvars